tests/fermi/test_fermionic.py

```diff
@@ -134,6 +134,9 @@ def test_to_mat(self):
         mat = fw1.to_mat()
         assert np.allclose(mat, expected_mat)
 
+        mat = fw1.to_mat(format="csr")
+        assert np.allclose(mat.toarray(), expected_mat)
+
```

Comment:
Are we not testing for when `format="dense"`? Also a little confused, it seems like there's no check for "csr" (also what does csr stand for? If it's not too long of a word I'd prefer to just spell it out in this case since it's not obviously to me what the acronym stands for).

Reply:
`csr` is just one type of sparse matrix that comes from [here](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_matrix.html).

Reply:
I see. Could maybe add a comment about it or not.

Reply:
The check when `format="dense"` happens with `mat = fw1.to_mat()` since "dense" is the default value. You're right that there is no check for "csr". This is because the method is just calling `to_mat()` on a `PauliSentence`, and that method just checks whether or not format is set to dense. All other values return a sparse matrix.

I think another test should be added that makes sure the type returned by `format="csr"` is a csr sparse matrix. Maybe that would also help clear up the meaning of csr?

Reply:
I see. To me it's a little strange that writing something like `format="whateveryouwant"` wouldn't raise an error. Are the only two choices "csr" and "dense" or could there be more supported matrix types?

Additionally, if a user intended to write "dense" but wrote "desne" by accident, this would not raise an error and just run as if the user wrote "csr". I think it's preferable to check for "dense" and "csr" specifically and raise a helpful error in all other cases.

Reply:
> I think another test should be added that makes sure the type returned by format="csr" is a csr sparse matrix. Maybe that would also help clear up the meaning of csr?

agree

Reply:
> I see. To me it's a little strange that writing something like `format="whateveryouwant"` wouldn't raise an error. Are the only two choices "csr" and "dense" or could there be more supported matrix types?
> 
> Additionally, if a user intended to write "dense" but wrote "desne" by accident, this would not raise an error and just run as if the user wrote "csr". I think it's preferable to check for "dense" and "csr" specifically and raise a helpful error in all other cases.

I agree, but this will require making changes to the `to_mat` method in `PauliSentence`. Right now the only two options are dense and csr. If anything other than "dense" is given, it will return a csr matrix. I could refactor it to raise errors if anything else is given, but maybe that should be a separate PR?

Reply:
Yeah maybe we don't want to raise the error right here, but rather in `PauliSentence`. If we're changing something in PauliSentence that should be a separate PR I think. We can wait for @soranjh to discuss more about this.